test/src/databases/all/functions.spec.ts

```diff
@@ -0,0 +1,1071 @@
+          ? []
+          : ([['floor(-1.9)', -2]] as [string, number][])),
+        ['floor(null)', null]
+      );
+      await funcTest('floor(1.9)', 1);
+    });
+  });
+
+  describe('ceil', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['ceil(1.9)', 2],
+        // TODO Remove when we upgrade to DuckDB 0.8.X -- DuckDB has some bugs with rounding
+        // that are fixed in 0.8.
+        ...(databaseName === 'duckdb_wasm'
+          ? []
+          : ([['ceil(-1.9)', -1]] as [string, number][])),
+        ['ceil(null)', null]
+      );
+    });
+  });
+
+  describe('length', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["length('foo')", 3], ['length(null)', null]);
+    });
+  });
+
+  describe('lower', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["lower('FoO')", 'foo'], ['lower(null)', null]);
+    });
+  });
+
+  describe('upper', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(["upper('fOo')", 'FOO'], ['upper(null)', null]);
+    });
+  });
+
+  describe('regexp_extract', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["regexp_extract('I have a dog', r'd[aeiou]g')", 'dog'],
+        ["regexp_extract(null, r'd[aeiou]g')", null],
+        ["regexp_extract('foo', null)", null]
+      );
+    });
+  });
+
+  describe('replace', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["replace('aaaa', 'a', 'c')", 'cccc'],
+        ["replace('aaaa', r'.', 'c')", 'cccc'],
+        [
+          "replace('axbxc', r'(a).(b).(c)', '\\\\0 - \\\\1 - \\\\2 - \\\\3')",
+          databaseName === 'postgres' ? '\\0 - a - b - c' : 'axbxc - a - b - c',
+        ],
+        ["replace('aaaa', '', 'c')", 'aaaa'],
+        ["replace(null, 'a', 'c')", null],
+        ["replace('aaaa', null, 'c')", null],
+        ["replace('aaaa', 'a', null)", null]
+      );
+    });
+  });
+
+  describe('substr', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ["substr('foo', 2)", 'oo'],
+        ["substr('foo', 2, 1)", 'o'],
+        ["substr('foo bar baz', -3)", 'baz'],
+        ['substr(null, 1, 2)', null],
+        ["substr('aaaa', null, 1)", null],
+        ["substr('aaaa', 1, null)", null]
+      );
+    });
+  });
+
+  describe('raw function call', () => {
+    it(`works - ${databaseName}`, async () => {
+      await funcTestMultiple(
+        ['floor(cbrt!(27)::number)', 3],
+        ['floor(cbrt!number(27))', 3],
+        ["substr('foo bar baz', -3)", 'baz'],
+        ['substr(null, 1, 2)', null],
+        ["substr('aaaa', null, 1)", null],
+        ["substr('aaaa', 1, null)", null]
+      );
+    });
+  });
+
+  describe('stddev', () => {
+    // TODO symmetric aggregates don't work with custom aggregate functions in BQ currently
```

Comment:
Worth noting that custom agg functions don't support symmetric aggregates in BQ today. We error if you try to use an agg function in a case where symmetric aggregates are required.